### PR TITLE
[kubernetes] Add demo helm chart docs

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Adds new documentation for the Demo helm chart.

Related to https://github.com/open-telemetry/opentelemetry.io/issues/3005

### CODE REVIEW

File: content/en/docs/kubernetes/helm/demo.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+```sh
+kubectl port-forward svc/my-otel-demo-frontendproxy 8080:8080
+```
+
+Once the proxy is exposed, you can also visit the following paths
+
+| Component         | Path                             |
+| ----------------- | -------------------------------- |
+| Webstore          | http://localhost:8080/           |
```
Comment (svrnm) [inline content/en/docs/kubernetes/helm/demo.md:35]:
```suggestion
| Web store          | http://localhost:8080/           |
```

File: content/en/docs/kubernetes/helm/demo.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+```sh
+kubectl port-forward svc/my-otel-demo-otelcol 4318:4318
+```
+
+### Configuration
+
+The Demo helm chart's default `values.yaml` is ready to be installed. All
+components have had their memory limits tuned to optimize performance, which may
+cause issues if your cluster is not large enough. The entire installation is
+restricted to ~3.5 Gi of memory, but may use less.
```
Comment (svrnm) [inline content/en/docs/kubernetes/helm/demo.md:53]:
```suggestion
restricted to ~3.5 Gigabytes of memory, but may use less.
```

**Review by svrnm — COMMENTED** [review]
thanks

**Comment by svrnm** [comment]
@TylerHelmuth there is also https://opentelemetry.io/docs/demo/kubernetes-deployment/ mentioning the helm chart, I am OK with having both, but we could also consider linking from the one page to the other to simplify things

**Comment by TylerHelmuth** [comment]
@svrnm added a link

**Review by cartermp — APPROVED** [review]

**Comment by cartermp** [comment]
Meegin', these look great

**Pull request merged**